src/opensearch_sdk_py/actions/response_handlers.py

```diff
@@ -23,6 +23,8 @@ def register(self, request_id: int, handler: ResponseHandler) -> None:
         self[request_id] = handler
 
     def handle(self, response: OutboundMessageResponse, input: StreamInput = None) -> Optional[bytes]:
-        handler = self[response.request_id]
-        del self[response.request_id]
-        return handler.handle(response, input) if handler else None
+        if response.request_id in self:
+            handler = self[response.request_id]
+            del self[response.request_id]
```

Comment:
Double-checking, if `handle` fails, do we still want to remove from `self`?

Reply:
> Double-checking, if `handle` fails, do we still want to remove from `self`?

Yep.  These response handlers directly correspond to requests we sent out, so will only be sent once by OpenSearch. We'll never get that requestId sent again so leaving it in the dict is useless.